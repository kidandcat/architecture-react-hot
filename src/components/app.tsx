import * as React from "react";
import { Clock } from "./clock";
import { ColorChanger } from "./colorChanger";

export class App extends React.Component {
  render() {
    return (
      <div>
        <h1>Hello World!</h1>
        <ColorChanger />
        <div>
          Here you have a Clock coloured with <span>Color</span>:
          <Clock />
        </div>
      </div>
    );
  }
}
