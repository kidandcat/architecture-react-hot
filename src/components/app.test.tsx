import * as React from "react";
import { shallow } from "enzyme";

import { App } from "./app";

test("Render Hello World!", () => {
  const app = shallow(<App />);
  expect(app.contains("Hello World!")).toBe(true);
});
