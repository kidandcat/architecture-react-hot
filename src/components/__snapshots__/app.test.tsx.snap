// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Hello World!
        </h1>,
        <div>
          <ColorChanger />
          Here you have a Clock coloured with
           
          <span
            style={
              Object {
                "color": "black",
              }
            }
          >
            black
          </span>
          :
          <Clock />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World!",
        },
        "ref": null,
        "rendered": "Hello World!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ColorChanger />,
            "Here you have a Clock coloured with",
            " ",
            <span
              style={
                Object {
                  "color": "black",
                }
              }
            >
              black
            </span>,
            ":",
            <Clock />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Here you have a Clock coloured with",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "black",
              "style": Object {
                "color": "black",
              },
            },
            "ref": null,
            "rendered": "black",
            "type": "span",
          },
          ":",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Hello World!
          </h1>,
          <div>
            <ColorChanger />
            Here you have a Clock coloured with
             
            <span
              style={
                Object {
                  "color": "black",
                }
              }
            >
              black
            </span>
            :
            <Clock />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello World!",
          },
          "ref": null,
          "rendered": "Hello World!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ColorChanger />,
              "Here you have a Clock coloured with",
              " ",
              <span
                style={
                  Object {
                    "color": "black",
                  }
                }
              >
                black
              </span>,
              ":",
              <Clock />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Here you have a Clock coloured with",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "black",
                "style": Object {
                  "color": "black",
                },
              },
              "ref": null,
              "rendered": "black",
              "type": "span",
            },
            ":",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
